### ISSUE TITLE
Discrepancies between paper and released training code (Stage 1 & 2)

### ISSUE THREAD

**Issue by zzh8241102** [YU-deep/vismem#12]
Hi, thanks so much for the great work and for releasing the code!  

While reproducing the training pipeline, I noticed a few` key discrepancies ` between the paper's description and the released code. I'd appreciate any clarification on whether these are intentional simplifications or if a more complete version is planned.

### Stage 1: Memory Formation
1. Training objective: SFT instead of GRPO
The paper describes Stage 1 using a GRPO objective (Section 3.4, Eq. 7, Appendix 7.2 Eq. 14, Table 4 with group_size=16). However, the released code uses a differential SFT loss without any reward computation or group sampling:
```
# train_stage1.py L71
loss = loss_mem - loss_base.detach()
```

   grpo.py is not imported by train_stage1.py.

2. Only short-term memory is trained
The paper states "we update the query builder B, and memory formers F{s/l}" and "randomly invoke either short- or long-term memory upon detecting the delimiter" (Section 3.4). However, the code hardcodes short-term only:
```
# stage1_memory_formation.py L27
M = vismem_model.form_memory(H, mem_type="short")  # or "long" / both, configurable -- only implemented short here
```
**Since Stage 2 freezes all memory formation parameters, the long_former is never trained in the entire pipeline.**

###  Stage 2: Invocation Optimization
I also noticed a few simplifications in Stage 2 compared to the paper:
* No group sampling: Paper specifies `group_size=16`, but the code generates only 1 sample per prompt. With a single sample, the group-relative advantage in SimpleGRPOTrainer is always 0 `(adv = rewards - rewards.mean() = 0),` so the policy gradient signal vanishes.
* No PPO-style clipping: Paper specifies `clip_ratio=0.2 (Table 4)`, but the code uses vanilla REINFORCE without importance ratio or clipping.
* Possible indentation issue in` train_stage2.py`: Lines 145–154 (gradient update, loss.backward(), opt.step()) appear to be outside the inner for i in pbar loop?

Are these simplifications in the released code intentional, or is there a plan to release the GRPO-based training pipeline described in the paper? Thanks in advance!

**Comment by YU-deep** [comment]
Hi, thank you for your attention to our work,  and we plan to release the complete GRPO-based training pipeline in the near future. And we appreciate the bugs you mentioned, which we will fix as soon as possible.
